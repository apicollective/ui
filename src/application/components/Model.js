// @flow
import React from 'react';

import H1 from 'components/H1';
import Markdown from 'components/Markdown';
import JsonDoc from 'application/components/JsonDoc/JsonDoc';
import ParameterListGroup from 'application/components/ParameterListGroup';

import { simplifyName } from 'utils';

import type {
  Service,
  Model as ModelType,
} from 'generated/version/ServiceType';

import styles from 'application/components/model.css';

const Model = ({
  model,
  service,
  importedServices,
  showJsonDoc,
}: {
  model: ModelType,
  service: Service,
  importedServices: Service[],
  showJsonDoc: boolean,
}) => (
  <div className={styles.container}>
    <div className={styles.header}>
      <H1 className={styles.title}>{simplifyName(model.name)}</H1>
      <Markdown
        source={model.description ? model.description : ''}
        className={styles.description}
      />
    </div>
    <div className={styles.fieldsContainer}>
      <ParameterListGroup
        parameters={model.fields}
        title="Fields"
        service={service}
        importedServices={importedServices}
        parentModel={model.name}
      />
      <div className={styles.json}>
        {showJsonDoc
          ? <JsonDoc
              baseModel={model.name}
              service={service}
              importedServices={importedServices}
              includeModel={false}
            />
          : null}
      </div>
    </div>
  </div>
);

export default Model;
