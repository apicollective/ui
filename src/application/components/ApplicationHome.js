// @flow
import React from 'react';

import { cleanPath } from 'utils';

import H1 from 'components/H1';
import Markdown from 'components/Markdown';
import ResourceCard from 'application/components/ResourceCard';

import type { Service } from 'generated/version/ServiceType';

import styles from 'application/components/application-home.css';

const ApplicationHome = ({
  service,
  organizationKey,
  applicationKey,
}: {
  service: Service,
  organizationKey: string,
  applicationKey: string,
}) => {
  const buildClickHref = (type, method, path) =>
    `/org/${organizationKey}/app/${applicationKey}/r/${type}/m/${method.toLowerCase()}/p/${cleanPath(path)}`;

  return (
    <div className={styles.content}>
      <div className={styles.header}>
        <H1 className={styles.h1}>{service.name}</H1>
        <Markdown
          source={service.description ? service.description : ''}
          className={styles.description}
        />
      </div>
      <div>
        {service.resources.map(resource =>
          resource.operations.map(operation => (
            <ResourceCard
              key={operation.method + operation.path}
              method={operation.method}
              path={operation.path}
              toHref={buildClickHref(
                resource.type,
                operation.method,
                operation.path
              )}
              description={operation.description}
            />
          ))
        )}
      </div>
    </div>
  );
};

export default ApplicationHome;
