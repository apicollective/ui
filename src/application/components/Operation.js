// @flow
import React from 'react';
import { Link } from 'utils';
import Markdown from 'components/Markdown';
import Header from 'application/components/Header';
import Request from 'application/components/Request';
import Response from 'application/components/Response';
import ResourceCard from 'application/components/ResourceCard';
import H1 from 'components/H1';

import type {
  Operation as OperationServiceType,
  Service,
  Resource as ResourceServiceType,
} from 'generated/version/ServiceType';

import styles from 'application/components/operation.css';

const Operation = ({
  service,
  operation,
  applicationKey,
  organizationKey,
  resource,
  method,
  path,
  importedServices,
}: {
  service: Service,
  operation: OperationServiceType,
  applicationKey: string,
  organizationKey: string,
  resource: ResourceServiceType,
  method: string,
  path: string,
  importedServices: Service[],
}) => (
  <div className={styles.content}>
    <div className={styles.header}>
      <H1 className={styles.h1}>
        <Link
          className={styles.link}
          to={`/org/${organizationKey}/app/${applicationKey}/m/${resource.type}`}
        >
          {resource.type}
        </Link>
      </H1>
      {resource.description &&
        <Markdown
          source={resource.description}
          className={styles.description}
        />}
    </div>
    <div className={styles.resource}>
      <ResourceCard method={operation.method} path={operation.path} />
    </div>
    {operation.description &&
      <Markdown
        source={operation.description}
        className={styles.description}
      />}
    <div className={styles.headers}>
      <Header
        appKey={applicationKey}
        orgKey={organizationKey}
        key={`${method}${resource.type}${path}-header`}
        operation={operation}
        service={service}
        importedServices={importedServices}
      />
    </div>
    <div className={styles.request}>
      <Request
        appKey={applicationKey}
        orgKey={organizationKey}
        key={`${method}${resource.type}${path}-request`}
        operation={operation}
        service={service}
        importedServices={importedServices}
      />
    </div>
    <Response
      appKey={applicationKey}
      orgKey={organizationKey}
      key={`${method}${resource.type}${path}-response`}
      operation={operation}
      service={service}
      importedServices={importedServices}
    />
  </div>
);

export default Operation;
