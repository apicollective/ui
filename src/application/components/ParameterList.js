// @flow
import React from 'react';
import classnames from 'classnames';

import Markdown from 'components/Markdown';
import {
  buildNavHref,
  getType,
  isImport,
  isImportOrInService,
  simplifyName,
  Link,
} from 'utils';

import type { Service } from 'generated/version/ServiceType';

import styles from 'application/components/parameter-list.css';

const ParameterList = (
  {
    name,
    type,
    required,
    description,
    minimum,
    maximum,
    example,
    defaultValue,
    service,
    importedServices,
    parentModel,
  }: {
    name: string,
    type: string,
    required: boolean,
    minimum?: number,
    maximum?: number,
    description?: string,
    example?: string,
    defaultValue?: string,
    service: Service,
    importedServices: Service[],
    parentModel: string,
  } = {}
) => {
  const possibleImportType = `${parentModel.substring(0, parentModel.lastIndexOf('.'))}.${type}`;
  const modelType = isImport(possibleImportType, importedServices)
    ? possibleImportType
    : type;
  const typeToHrefFn = isImportOrInService(
    getType(modelType),
    service,
    importedServices
  )
    ? buildNavHref({
        organization: service.organization.key,
        application: service.application.key,
        model: getType(modelType),
      })
    : null;

  return (
    <div className={styles.container}>
      <div className={styles.meta}>
        <a name={`${getType(parentModel)}.${name}`} className={styles.name}>
          {name}
        </a>
        {/* <p onClick={typeClickFn} className={classnames(styles.type, typeClickFn ? styles.pointer : null)}>
            {simplifyName(modelType)}
            </p> */}
        <Link
          tabIndex={0}
          toHref={typeToHrefFn}
          className={classnames(styles.type)}
        >
          {simplifyName(modelType)}
        </Link>
        {required ? <p className={styles.required}>required</p> : null}
      </div>
      <div className={styles.info}>
        {description
          ? <Markdown source={description} className={styles.description} />
          : <p className={styles.noContent}>No description</p>}
        {minimum
          ? <p className={styles.sample}>
              <span className={styles.sampleTitle}>Minimum</span>{minimum}
            </p>
          : null}
        {maximum
          ? <p className={styles.sample}>
              <span className={styles.sampleTitle}>Maximum</span>{maximum}
            </p>
          : null}
        {example
          ? <p className={styles.sample}>
              <span className={styles.sampleTitle}>Example</span>{example}
            </p>
          : null}
        {defaultValue
          ? <p className={styles.sample}>
              <span className={styles.sampleTitle}>Default</span>{defaultValue}
            </p>
          : null}
      </div>
    </div>
  );
};

export default ParameterList;
