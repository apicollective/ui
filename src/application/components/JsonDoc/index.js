import React, { Component, PropTypes } from 'react';

import styles from './jsonDoc.css';
import H2 from '../../../components/H2';
import ParameterList from '../ParameterList';

const numSpaces = 4;

const spaces = (indent) => Array(indent * numSpaces).join(' ');

const getType = (type) => {
  const ex = /[\[]?([^\]]+)/i;
  return type.match(ex)[1];
};

const getModel = (name, spec) => spec.models.find(m => m.name === getType(name));
const isModel = (type, spec) => Boolean(getModel(type, spec));

const getEnum = (type, spec) => spec.enums.find(e => e.name === type);
const isEnum = (type, spec) => Boolean(getEnum(type, spec));

const isArray = (type) => type.startsWith('[');


// --    "name": "value",
/* eslint-disable max-len */
const FieldValue = ({ name, value, fullType, indent, mouseOver }) =>
  <a href={`#${fullType}`} className={styles.link} data-fullType={fullType} onMouseOver={mouseOver}>
    <span className={styles.name}>{spaces(indent)}"{name}"</span>: <span className={styles.value}>"{value}"</span>,{`\n`}
  </a>;
/* eslint-enable */

FieldValue.propTypes = {
  name: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  fullType: PropTypes.string.isRequired,
  indent: PropTypes.number.isRequired,
  mouseOver: PropTypes.func.isRequired,
};

// --    "name":
const FieldEmpty = ({ name, fullType, indent, mouseOver }) =>
  <a href={`#${fullType}`} className={styles.link} data-fullType={fullType} onMouseOver={mouseOver}>
    <span className={styles.name}>{spaces(indent)}"{name}"</span>:
  </a>;

FieldEmpty.propTypes = {
  name: PropTypes.string.isRequired,
  fullType: PropTypes.string.isRequired,
  indent: PropTypes.number.isRequired,
  mouseOver: PropTypes.func.isRequired,
};

// --   "value",
const StringValue = ({ value, fullType, indent, mouseOver }) =>
  <a href={`#${fullType}`} className={styles.link} data-fullType={fullType} onMouseOver={mouseOver}>
    <span className={styles.name}>{spaces(indent)}"{value}"</span>{`\n`}
  </a>;

StringValue.propTypes = {
  value: PropTypes.string.isRequired,
  fullType: PropTypes.string.isRequired,
  indent: PropTypes.number.isRequired,
  mouseOver: PropTypes.func.isRequired,
};

const ArrayValue = ({ name, fullType, indent, mouseOver }) =>
  <div>
    <FieldEmpty name={name} fullType={fullType} indent={indent} mouseOver={mouseOver} /> {'[\n'}
    <StringValue value={name} fullType={fullType} indent={indent + 1} mouseOver={mouseOver} />
    {`${spaces(indent)}],`}
  </div>;

ArrayValue.propTypes = {
  name: PropTypes.string.isRequired,
  fullType: PropTypes.string.isRequired,
  indent: PropTypes.number.isRequired,
  mouseOver: PropTypes.func.isRequired,
};

const renderModel = (key, name, type, fullType, spec, indent, mouseOver) => {
  let open = '{';
  let close = '}';
  if (isArray(type)) {
    open = '[{';
    close = '}]';
  } else if (isEnum(type, spec)) {
    open = '[';
    close = ']';
  }
  return (
    <Model
      key={key}
      name={name}
      type={type}
      fullType={fullType}
      spec={spec}
      indent={indent}
      mouseOver={mouseOver}
      open={open}
      close={close}
    />
  );
};

const ModelInner = ({ type, fullType, spec, indent, mouseOver }) => {
  if (isEnum(type, spec)) {
    // Enum Value
    const enumModel = getEnum(type, spec);
    return (
      <StringValue value={enumModel.fields[0].name} fullType={fullType} indent={indent + 1} mouseOver={mouseOver} />
    );
  } else {
    // Model, Array, Field
    return (
      <div>
        {getModel(type, spec).fields.map((field, id) => {
          const value = field.example;
          if (isModel(field.type, spec)) {
            return renderModel(id, field.name, field.type, `${type}.${field.name}`, spec, indent + 1, mouseOver);
          } else if (isArray(field.type)) {
            return (
              <ArrayValue
                key={id}
                name={field.name}
                value={value}
                fullType={`${type}.${field.name}`}
                indent={indent + 1}
                mouseOver={mouseOver}
              />);
          } else {
            return (
              <FieldValue
                key={id}
                name={field.name}
                value={value}
                fullType={`${type}.${field.name}`}
                indent={indent + 1}
                mouseOver={mouseOver}
              />);
          }
        })}
      </div>
    );
  }
};
ModelInner.propTypes = {
  type: PropTypes.string.isRequired,
  fullType: PropTypes.string,
  spec: PropTypes.object.isRequired,
  indent: PropTypes.number.isRequired,
  mouseOver: PropTypes.func.isRequired,
};

const Model = ({ name, type, fullType, spec, indent, mouseOver, open, close }) =>
  <div>
    <FieldEmpty name={name} fullType={fullType} indent={indent} mouseOver={mouseOver} /> {open}
    <ModelInner type={getType(type)} fullType={fullType} spec={spec} indent={indent} mouseOver={mouseOver} />
    {`${spaces(indent)}${close},`}
  </div>;

Model.propTypes = {
  name: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  fullType: PropTypes.string.isRequired,
  spec: PropTypes.object.isRequired,
  indent: PropTypes.number.isRequired,
  mouseOver: PropTypes.func.isRequired,
  open: PropTypes.string.isRequired,
  close: PropTypes.string.isRequired,
};


const Documentation = ({ field }) =>
  <div className={styles.documentation}>
    <ParameterList {...field} />
  </div>;

Documentation.propTypes = {
  field: PropTypes.object.isRequired,
};

// TODO
// Support for markdown descriptions
class JsonDoc extends Component {
  constructor(props) {
    super(props);
    this.state = { documentationFullType: '' };

    this.mouseOver = (event) => {
      const documentationFullType = event.currentTarget.dataset.fulltype;
      this.setState({ documentationFullType });
    };
  }

  getDocumentation(documentationFullType, spec) {
    if (!documentationFullType) return '';

    const [modelName, fieldName] = documentationFullType.split('.');
    const model = getModel(modelName, spec);
    const field = model.fields.find(f => f.name === fieldName);

    return <Documentation field={field} />;
  }

  getModelDoc(baseModel, spec, includeModel) {
    const docs = () => {
      const model = getModel(baseModel, spec);
      return (
        <div>
          <H2 className={styles.modelName}>{baseModel}</H2>
          <p className={styles.modelDescription}>{model.description}</p>
        </div>
      );
    };
    return includeModel ? docs() : null;
  }

  render() {
    const { baseModel, spec, includeModel } = this.props;
    return (
      <div className={styles.jsonDoc}>
        {this.getModelDoc(baseModel, spec, includeModel)}
        <div className={styles.container}>
          <pre className={styles.code}>
          {'{'}
            <ModelInner
              type={baseModel}
              spec={spec}
              indent={0}
              mouseOver={this.mouseOver}
            />
          {'}'}
          </pre>
          {this.getDocumentation(this.state.documentationFullType, spec)}
        </div>
      </div>
    );
  }
}
JsonDoc.propTypes = {
  spec: PropTypes.object.isRequired,
  baseModel: PropTypes.string.isRequired,
  includeModel: PropTypes.bool, // Include Model Documentation in JsonDoc
};

export default JsonDoc;
