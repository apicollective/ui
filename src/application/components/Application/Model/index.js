import React, { PropTypes } from 'react';

import H1 from '../../../../components/H1';
import H2 from '../../../../components/H2';
import JsonDoc from '../../JsonDoc';
import ParameterList from '../../ParameterList';

import styles from './model.css';

const Model = ({ modelName, spec }) => {
  const model = spec.models.find(m => m.name === modelName);
  return (
    <div>
      <H1>{model.name}</H1>
      {model.description ? <p className={styles.description}>{model.description}</p> : null}
      <H2>Fields</H2>
      {model.fields.map((field, id) => (
        <ParameterList {...field} />
      ))}
      <JsonDoc baseModel={model.name} spec={spec} includeModel={false} />
    </div>
  );
};
Model.propTypes = {
  modelName: PropTypes.string.isRequired,
  spec: PropTypes.object.isRequired,
};

export default Model;
