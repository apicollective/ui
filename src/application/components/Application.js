// @flow
import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import LoadingOverlay from 'components/LoadingOverlay';
import Operation from 'application/components/Operation';
import ApplicationHome from 'application/components/ApplicationHome';
import * as utils from 'utils';
import Model from 'application/components/Model';
import Enum from 'application/components/Enum';

import { actions as serviceActions } from 'generated/version';

import type { Service } from 'generated/version/ServiceType';
import type { Match } from 'params';

import styles from 'application/components/application.css';

const allActions = Object.assign({}, serviceActions);

type Params = {|
  model?: string,
  resource?: string,
  method: string,
  path: string,
  applicationKey: string,
  organizationKey: string,
|};

type Props = {|
  actions: Object, // FIXME
  match: Match<Params>,
  loaded: boolean,
  service: Service,
  importedServices: Service[],
|};

export class Application extends Component {
  props: Props;

  componentDidMount() {
    const orgKey = this.props.match.params.organizationKey;
    const applicationKey = this.props.match.params.applicationKey;
    this.props.actions.getByOrgkeyAndApplicationkeyAndVersion_get({
      orgKey,
      applicationKey,
      version: 'latest',
    });
  }

  render() {
    const { service, importedServices } = this.props;

    if (!service) {
      return <LoadingOverlay isLoaded={this.props.loaded} />;
    }
    if (this.props.match.params.resource) {
      // Load Operation
      const {
        method,
        path,
        resource: resourceType,
        applicationKey,
        organizationKey,
      } = this.props.match.params;

      const operation = utils.getOperation(resourceType, method, path, service);
      const resource = utils.getResource(resourceType, service);

      return (
        <LoadingOverlay isLoaded={this.props.loaded}>
          <Operation
            service={service}
            importedServices={importedServices}
            operation={operation}
            applicationKey={applicationKey}
            organizationKey={organizationKey}
            resource={resource}
            method={method}
            path={path}
          />
        </LoadingOverlay>
      );
    } else if (this.props.match.params.model) {
      // Load Model
      const modelName = this.props.match.params.model;
      if (utils.isEnum(modelName, service, importedServices)) {
        const enumModel = utils.getEnum(modelName, service, importedServices);
        return enumModel
          ? <LoadingOverlay isLoaded={this.props.loaded}>
              <Enum
                enumModel={enumModel}
                service={service}
                importedServices={importedServices}
              />
            </LoadingOverlay>
          : null;
      } else {
        const model = utils.getModel(modelName, service, importedServices);
        return model
          ? <LoadingOverlay isLoaded={this.props.loaded}>
              <Model
                model={model}
                service={service}
                importedServices={importedServices}
                showJsonDoc={true}
              />
            </LoadingOverlay>
          : null;
      }
    } else {
      // Load Application Home
      const { applicationKey, organizationKey } = this.props.match.params;

      return (
        <LoadingOverlay isLoaded={this.props.loaded}>
          <ApplicationHome
            service={service}
            applicationKey={applicationKey}
            organizationKey={organizationKey}
          />
        </LoadingOverlay>
      );
    }
  }
}

const mapStateToProps = state => ({
  loaded: state.application.loaded,
  service: state.application.service,
  importedServices: state.application.importedServices,
});

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators(allActions, dispatch),
});

export default connect(mapStateToProps, mapDispatchToProps)(Application);

export { styles };
