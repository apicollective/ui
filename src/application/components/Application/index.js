import React, { Component, PropTypes } from 'react';
import classnames from 'classnames';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import JsonDoc from './../JsonDoc';
import H1 from '../../../components/H1';
import H2 from '../../../components/H2';
import ParameterList from '../ParameterList';
import { cleanPath } from '../../../utils';
import Model from './Model';

import styles from './application.css';

import { actions as specActions } from '../../../generated/version';
const allActions = Object.assign({}, specActions);

const Request = ({ operation, spec }) => {
  const body = () => {
    if (operation.body) {
      const baseModel = operation.body.type;
      return (
        <div>
          <h3>Body</h3>
          <JsonDoc baseModel={baseModel} spec={spec} />
        </div>
      );
    } else return null;
  };

  return (
    <div className={classnames(styles.section, styles.request)}>
      <H2 className={styles.sectionHeader}>Request</H2>
      {operation.parameters.map((param, id) => (
        <ParameterList {...param} />
      ))}
    {body()}
    </div>
  );
};
Request.propTypes = {
  operation: PropTypes.object.isRequired,
  spec: PropTypes.object.isRequired,
};

const Response = ({ operation, spec }) => {
  const body = (response) => {
    if (response.type) {
      const baseModel = response.type;
      return (
        <div>
          <h3>Body</h3>
          <JsonDoc baseModel={baseModel} spec={spec} includeModel={true} />
        </div>
      );
    } else return null;
  };

  return (
    <div className={classnames(styles.section, styles.response)}>
      <H2>Response</H2>
      {operation.responses.map((response, id) => (
        <div key={id}>
          <div>{response.code.integer.value}</div>
          {body(response)}
        </div>
      ))}
    </div>
  );
};
Response.propTypes = {
  operation: PropTypes.object.isRequired,
  spec: PropTypes.object.isRequired,
};


class Application extends Component {

  componentDidMount() {
    const orgKey = this.props.params.organizationKey;
    const applicationKey = this.props.params.applicationKey;
    this.props.actions.getByOrgkeyAndApplicationkeyAndVersion_get(
      { orgKey, applicationKey, version: 'latest' }
    );
  }

  getOperation(type, method, path, spec) {
    const resource = spec.resources.find(r => r.type === type);
    const operation = resource.operations.find((o) => (
      o.method.toLowerCase() === method && cleanPath(o.path) === path
    ));
    return operation;
  }

  render() {
    const { spec } = this.props;
    if (!this.props.loaded) {
      // First Load
      return (<div></div>);
    } else if (this.props.params.resource) {
      // Load Operation
      const { resource, method, path } = this.props.params;
      const operation = this.getOperation(resource, method, path, spec);
      return (
        <div>
          <div className={styles.specHeader}>
            <H1>{spec.name}</H1>
            <p className={styles.description}>{spec.description}</p>
          </div>
          <H2>{operation.method} {operation.path}</H2>
          <Request operation={operation} spec={spec} />
          <Response operation={operation} spec={spec} />
        </div>
      );
    } else if (this.props.params.model) {
      return <Model modelName={this.props.params.model} spec={spec} />;
    } else {
      // No Operation
      return (
        <div>
          <div className={styles.specHeader}>
            <H1>{spec.name}</H1>
            <p className={styles.description}>{spec.description}</p>
          </div>
          <div>
            {spec.resources.map((resource, id) => (
              resource.operations.map((operation, resourceId) => (
                <div>{operation.method} {operation.path}</div>
              ))
            ))}
          </div>
        </div>
      );
    }
  }
}
Application.propTypes = {
  actions: PropTypes.object.isRequired,
  params: PropTypes.object.isRequired,
  loaded: PropTypes.bool.isRequired,
  spec: PropTypes.object.isRequired,
};

const mapStateToProps = (state) => (
  {
    loaded: state.application.get('loaded'),
    spec: state.application.get('spec'),
  }
);

const mapDispatchToProps = (dispatch) => (
  { actions: bindActionCreators(allActions, dispatch) }
);

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Application);

export {
  styles,
};
