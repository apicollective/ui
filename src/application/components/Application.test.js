// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router';

import { Application } from 'application/components/Application';
import Example from 'exampleService.json';
import { actions as serviceActions } from 'generated/version';

test('Application Home', () => {
  const params = {
    actions: serviceActions,
    match: {
      params: {
        model: undefined,
        resource: undefined,
        method: '',
        path: '',
        applicationKey: '',
        organizationKey: '',
      },
      url: '/',
      path: '/',
      isExact: true,
    },
    loaded: true,
    service: Example,
    importedServices: [],
  };

  const component = renderer.create(
    <MemoryRouter>
      <Application {...params} />
    </MemoryRouter>
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
