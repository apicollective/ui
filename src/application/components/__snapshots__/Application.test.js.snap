// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application Home 1`] = `
<div>
  <div>
    <div
      className="content"
    >
      <div
        className="header"
      >
        <h1
          className="h1 h1"
        >
          apidoc generator
        </h1>
        <div
          className="description container"
        >
          <div
            className="markdown"
          >
            <p>
              Documentation for an apidoc code generator API
            </p>
          </div>
        </div>
      </div>
      <div>
        <div
          className="container href"
        >
          <a
            href="/org//app//r/generator/m/get/p/generators"
            onClick={[Function]}
            tabIndex={0}
          >
            <div
              className="flex"
            >
              <div
                className="left"
              >
                <div
                  className="method get isExpandedMethod"
                >
                  GET
                </div>
              </div>
              <div
                className="right"
              >
                <pre
                  className="path isExpandedPath"
                >
                  /generators
                </pre>
              </div>
            </div>
            <div
              className="description container"
            >
              <div
                className="markdown"
              >
                <p>
                  Get all available generators
                </p>
              </div>
            </div>
          </a>
        </div>
        <div
          className="container href"
        >
          <a
            href="/org//app//r/generator/m/get/p/generatorskey"
            onClick={[Function]}
            tabIndex={0}
          >
            <div
              className="flex"
            >
              <div
                className="left"
              >
                <div
                  className="method get isExpandedMethod"
                >
                  GET
                </div>
              </div>
              <div
                className="right"
              >
                <pre
                  className="path isExpandedPath"
                >
                  /generators/:key
                </pre>
              </div>
            </div>
            <div
              className="description container"
            >
              <div
                className="markdown"
              >
                <p>
                  Get generator with this key
                </p>
              </div>
            </div>
          </a>
        </div>
        <div
          className="container href"
        >
          <a
            href="/org//app//r/healthcheck/m/get/p/_internal_healthcheck"
            onClick={[Function]}
            tabIndex={0}
          >
            <div
              className="flex"
            >
              <div
                className="left"
              >
                <div
                  className="method get"
                >
                  GET
                </div>
              </div>
              <div
                className="right"
              >
                <pre
                  className="path"
                >
                  /_internal_/healthcheck
                </pre>
              </div>
            </div>
          </a>
        </div>
        <div
          className="container href"
        >
          <a
            href="/org//app//r/invocation/m/post/p/invocationskey"
            onClick={[Function]}
            tabIndex={0}
          >
            <div
              className="flex"
            >
              <div
                className="left"
              >
                <div
                  className="method post isExpandedMethod"
                >
                  POST
                </div>
              </div>
              <div
                className="right"
              >
                <pre
                  className="path isExpandedPath"
                >
                  /invocations/:key
                </pre>
              </div>
            </div>
            <div
              className="description container"
            >
              <div
                className="markdown"
              >
                <p>
                  Invoke a generator
                </p>
              </div>
            </div>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
